File: myapp/lib/services/googlesignin.php
<?php
header("Access-Control-Allow-Origin: *");
header("Content-Type: application/json; charset=UTF-8");

$data = json_decode(file_get_contents("php://input"));

if(!empty($data->idToken) && !empty($data->email)) {
    // Verify Google ID token (you should implement proper verification)
    // For now, we'll just use the received data
    
    $conn = new mysqli("localhost", "username", "password", "your_database");
    
    // Check if user exists
    $query = "SELECT * FROM users WHERE email = ?";
    $stmt = $conn->prepare($query);
    $stmt->bind_param("s", $data->email);
    $stmt->execute();
    $result = $stmt->get_result();
    
    if($result->num_rows === 0) {
        // User doesn't exist, create new account
        $insertQuery = "INSERT INTO users (google_id, name, email, profile_pic) VALUES (?, ?, ?, ?)";
        $insertStmt = $conn->prepare($insertQuery);
        $insertStmt->bind_param("ssss", $data->googleId, $data->name, $data->email, $data->photoUrl);
        $insertStmt->execute();
        $userId = $conn->insert_id;
    } else {
        // User exists, get user data
        $user = $result->fetch_assoc();
        $userId = $user['id'];
    }
    
    // Return user data
    echo json_encode([
        'status' => 'success',
        'user' => [
            'id' => $userId,
            'name' => $data->name,
            'email' => $data->email,
            'profile_pic' => $data->photoUrl
        ]
    ]);
} else {
    echo json_encode(['status' => 'error', 'message' => 'Invalid data']);
}
?>

File: myapp/lib/services/googlesignin.dart
import 'package:google_sign_in/google_sign_in.dart';
import 'package:http/http.dart' as http;
import 'dart:convert';

class GoogleSignInService {
  static final GoogleSignIn _googleSignIn = GoogleSignIn(
    scopes: [
      'email',
      'profile',
    ],
  );

  static Future<GoogleSignInAccount?> signInWithGoogle() async {
    try {
      final account = await _googleSignIn.signIn();
      if (account != null) {
        return account;
      }
      return null;
    } catch (error) {
      print('Google Sign-In Error: $error');
      return null;
    }
  }

  static Future<void> signOut() async {
    await _googleSignIn.signOut();
  }

  static Future<Map<String, dynamic>?> signInAndSendToBackend() async {
    try {
      final account = await signInWithGoogle();
      if (account == null) {
        return null;
      }
      final auth = await account.authentication;

      final response = await http.post(
        Uri.parse(' https://db37-2407-d000-d-33c2-15fd-c4ba-2d0-db4d.ngrok-free.app'), // <-- your ngrok URL
        headers: {
          'Content-Type': 'application/json',
        },
        body: jsonEncode({
          'idToken': auth.idToken,
          'googleId': account.id,
          'name': account.displayName,
          'email': account.email,
          'photoUrl': account.photoUrl,
        }),
      );

      if (response.statusCode == 200) {
        return jsonDecode(response.body);
      } else {
        return {'status': 'error', 'message': 'Server error'};
      }
    } catch (e) {
      print('Google Sign-In Error: $e');
      return null;
    }
  }
}